components/ImageButton/ImageButton.js

```diff
@@ -0,0 +1,45 @@
+
+import React, { PureComponent } from 'react';
+import { TouchableOpacity, StyleSheet } from 'react-native';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import { connect } from 'react-redux';
+import { addModal } from 'actions';
+
+type Props = {
+    children: React$Element,
+    onPress: () => any,
+    style?: ComponentStyles,
+    addModal: typeof addModal
+};
+
+class ImageButton extends PureComponent<Props> {
+    render() {
+        const { onPress } = this.props;
+        const image = 'ellipsis-h';
+        return (
+            <TouchableOpacity
+              onPress={onPress}
+              style={styles.imageBackground}>
+              <Icon name={image} size={20} />
+            </TouchableOpacity>
+        );
+    }
+}
+
+export default connect(
+  null,
+  { addModal }
+)(ImageButton);
```

Comment:
`connect`ing a component to the redux store w/o props passing is the wrong way.
Bind `dispatch` to `addModal` where you're rendering this component pass it as a prop here. But seems like you don't use it here, so remove `connect`ing at all.